modules/internal/ChapelIteratorSupport.chpl

```diff
@@ -17,15 +17,35 @@
  * limitations under the License.
  */
 
+// This comment is a lie for chpl-docs sake. It only applies to the
+// vectorizeOnly iterators found at the bottom of this file.
+/*
+  Data parallel constructs (such as ``forall`` loops) are implicitly
+  vectorizable. If the ``--vectorize`` compiler flag is thrown (implied by
+  ``--fast``), the Chapel compiler will emit vectorization hints to the backend
+  compiler.
```

Comment:
Should we note something about "for supported compilers" here?  If we already list these compilers elsewhere (man page?) cross-reference to avoid documentation maintenance issues?